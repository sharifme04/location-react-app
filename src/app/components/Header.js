/* eslint-disable jsx-a11y/anchor-is-valid */
import React from "react";
import PropTypes from "prop-types";

const Header = ({
  title,
  subt1,
  subt2,
  subt3,
  subt4,
  subt5,
  subt6,
  subt7,
  subt8,
}) => {
  return (
    <div>
      <div>
        <span className="title">{title}</span>
      </div>
      <div className="subtitle">
        {subt1 && <span>{subt1}</span>}
        {subt2 && <span ><a  className="link-subtitle" href="#">{subt2}</a> </span>}
        {subt3 && <span>{subt3}</span>}
        {subt4 && <span ><a  className="link-subtitle" href="#">{subt4}</a> </span>}
        {subt5 && <span>{subt5}</span>}
        {subt6 && <span ><a  className="link-subtitle" href="#">{subt6}</a> </span>}
        {subt7 && <span>{subt7}</span>}
        {subt8 && <span ><a  className="link-subtitle" href="#">{subt8}</a> </span>}
      </div>
    </div>
  );
};

Header.propTypes = {
    title: PropTypes.string,
    subt1: PropTypes.string,
    subt2: PropTypes.string,
    subt3: PropTypes.string,
    subt4: PropTypes.string,
    subt5: PropTypes.string,
    subt6: PropTypes.string,
    subt7: PropTypes.string,
    subt8: PropTypes.string,
};

export default Header;
