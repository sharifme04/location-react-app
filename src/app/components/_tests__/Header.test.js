import React from "react";
import renderer from "react-test-renderer";
import Header from "../Header";

test("snapshot Form renders with empty props", () => {
  const component = renderer.create(<Header />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot Form renders with  props", () => {
  const component = renderer.create(
    <Header
      subt1={"Umwelt und Preisvergleich zwischen VVS und Deinem Auto"}
      title={"Umweltrechner"}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
