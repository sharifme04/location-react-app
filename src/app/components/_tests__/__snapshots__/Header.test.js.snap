// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Form renders with  props 1`] = `
<div>
  <div>
    <span
      className="title"
    >
      Umweltrechner
    </span>
  </div>
  <div
    className="subtitle"
  >
    <span>
      Umwelt und Preisvergleich zwischen VVS und Deinem Auto
    </span>
  </div>
</div>
`;

exports[`snapshot Form renders with empty props 1`] = `
<div>
  <div>
    <span
      className="title"
    />
  </div>
  <div
    className="subtitle"
  />
</div>
`;
